host/crypto/asn1.c

```diff
@@ -44,7 +44,7 @@ oe_result_t oe_asn1_get_raw(
             &tmp_length,
             &tmp_tag,
             &tmp_class,
-            oe_asn1_remaining(asn1));
+            (long)oe_asn1_remaining(asn1));
```

Comment:
A sign-conversion warning occurs here (conversion to `long int` from `size_t`)

Reply:
So this is actually an unsafe cast because of the sign conversion; we need to add a check for the return value of `oe_asn1_remaining` against `OE_INT64_MAX`. This also applies elsewhere like on line 135 below.

Reply:
The return of `oe_asn1_remaining` is a difference of two addresses casted explicitly to `size_t`.  Wondering whether the size check is necessary.